src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12163,23 +12179,13 @@ void Unit::UpdateObjectVisibility(bool forced)
 
 void Unit::KnockbackFrom(float x, float y, float speedXY, float speedZ)
 {
-    Player* player = ToPlayer();
-    if (!player)
-    {
-        if (Unit* charmer = GetCharmer())
-        {
-            player = charmer->ToPlayer();
-            if (player && player->GetUnitBeingMoved() != this)
-                player = nullptr;
-        }
-    }
-
-    if (!player)
+    if (IsMovedByServer())
     {
         GetMotionMaster()->MoveKnockbackFrom(x, y, speedXY, speedZ);
     }
     else
     {
+        Player* playerMover = GetGameClientMovingMe()->GetBasePlayer();
```

Comment:
so, is a "mover" the player moving me...?

Reply:
Here the naming is bad. that Player unit might not even be moving. I changed it to basePlayer. We only need this reference to send the message to a specific client. It might be more readable if I were to add `SendDirectMessage` to GameClient as well

Reply:
done